kuma/users/tests/test_views.py

```diff
@@ -1295,26 +1297,203 @@ def test_account_refresh_token_saved_next_login(self):
+        revision=revision, sender=wiki_user)
+
+    attachment_revision = AttachmentRevision(
+        attachment=Attachment.objects.create(title='test attachment'),
+        file='some/path.ext',
+        mime_type='application/kuma',
+        creator=wiki_user,
+        title='test attachment',
+    )
+    attachment_revision.save()
+    assert AttachmentRevision.objects.filter(creator=wiki_user).exists()
+
+    url = reverse('users.user_delete', kwargs={'username': wiki_user.username})
+    response = user_client.post(url, {'attributions': 'donate'}, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 302
+    assert not User.objects.filter(username=wiki_user.username).exists()
+
+    revision.refresh_from_db()
+    assert revision.creator.username == 'Anonymous'
+
+    attachment_revision.refresh_from_db()
+    assert attachment_revision.creator.username == 'Anonymous'
+
+    # The user_client should now become "invalid" since its session
+    # is going to point to no user.
+    response = user_client.get(url, HTTP_HOST=settings.WIKI_HOST)
+    assert response.status_code == 302
+    assert reverse('account_login') in response['Location']
+    # Let's doublecheck that
+    whoami_url = reverse('api.v1.whoami')
+    response = user_client.get(whoami_url)
+    assert response.status_code == 200
+    assert not response.json()['username']
+    assert not response.json()['is_authenticated']
+
+
+def test_delete_user_keep_attributions(db, user_client, wiki_user):
+
+    # Pretend the user logged in with GitHub
+    SocialAccount.objects.create(
+        user=wiki_user,
+        provider='github',
+        extra_data=dict(
+            email=wiki_user.email,
+            avatar_url='https://avatars0.githubusercontent.com/yada/yada',
+            html_url="https://github.com/{}".format(wiki_user.username)
+        )
+    )
+
+    # Also, pretend that the user has a rich profile
+    User.objects.filter(id=wiki_user.id).update(
+        first_name='Peter',
+        last_name='B',
+        timezone='Ocean',
+        locale='sv-SE',
+        homepage='https://www.peterbe.com',
+        title='Web Dev',
+        fullname='Peter B',
+        organization='Mozilla',
+        location='Earth',
+        bio='Doing stuff',
+        irc_nickname='pb',
+        website_url='https://www.peterbe.com',
+        github_url='github/peterbe',
+        mozillians_url='mozillians/peterbe',
+        twitter_url='twitter/peterbe',
+        linkedin_url='linkedin/peterbe',
+        facebook_url='facebook/peterbe',
+        stackoverflow_url='stackoverflow/peterbe',
+        discourse_url='discourse/peterbe',
+        stripe_customer_id='123456',
+    )
+
+    document = create_document(save=True)
+    revision = create_revision(
+        title='My First And Only Revision',
+        document=document,
+        creator=wiki_user,
+        save=True)
+    assert Revision.objects.filter(creator=wiki_user).exists()
```

Comment:
Same here, you could use the `create_revision` pytest fixture instead of this chunk of code.

Reply:
I don't undertstand that fixture. It's a fixture that returns an instance but doesn't create anything. 
However, `from kuma.wiki.tests import revision as create_revision` gets me a function that creates Revisions.

Reply:
Oops, sorry, you could just use the `root_doc` fixture, which provides a doc and its revision.